tests/adapters/test_visa.py

```diff
@@ -88,14 +99,34 @@ def test_write_bytes_read(adapter):
     assert adapter.read() == "SCPI,MOCK,VERSION_1.0"
 
 
-def test_write_read_bytes(adapter):
-    adapter.write("*IDN?")
-    assert adapter.read_bytes(22) == b"SCPI,MOCK,VERSION_1.0\n"
-
-
-def test_write_read_all_bytes(adapter):
-    adapter.write("*IDN?")
-    assert adapter.read_bytes(-1) == b"SCPI,MOCK,VERSION_1.0\n"
+class TestReadBytes:
+    @pytest.fixture()
+    def adapter(self, adapter):
+        adapter.write("*IDN?")
+        yield adapter
+        # empty the read buffer
+        try:
+            adapter.read_bytes(-1)
+        except pyvisa.errors.VisaIOError as exc:
+            if not exc.args[0].startswith("VI_ERROR_TMO"):
+                raise
```

Comment:
Why did you not add the shutdown behaviour to the fixture at the top? 
Also, please give it a different name, this overloading is mighty confusing. :D
By the way, it turns out we need to `.close()` the pyvisa resource manager, then the buffer is correctly emptied (as advised in the upstream issue). I also realised that we are not cleaning this up properly, as the `close()` of our `VISAAdapter` does not call `close` on `self.manager`!

Reply:
`read_bytes` runs into a timeout, if there is no buffer.
To reduce the test time, I only clear the read buffer, where I test reading (which might fail).

Thanks for raising the issue with close. Maybe that solves the problem (and improves our adapter).

Reply:
I put the cleanup in the top fixture, together with "close".

Even with closing the manager (which I added, inclusive the appropriate tests), I have to use the cleanup procedure.
`Clear` is not implemented in the sim and flush buffer seems not to work (see comments).

Reply:
thanks, LGTM.